compiled_starters/cpp/src/main.cpp

```diff
@@ -0,0 +1,66 @@
+#include <cstdlib>
+#include <cstring>
+#include <iostream>
+#include <netdb.h>
+#include <string>
+#include <system_error>
+#include <unistd.h>
+#include <arpa/inet.h>
+#include <sys/socket.h>
+#include <sys/types.h>
+
+int main(int argc, char* argv[]) {
+    // Flush after every std::cout / std::cerr
+    std::cout << std::unitbuf;
+    std::cerr << std::unitbuf;
+
+    // You can use print statements as follows for debugging, they'll be visible when running tests.
+    std::cerr << "Logs from your program will appear here!\n";
+
+    // Create socket
+    int server_fd = socket(AF_INET, SOCK_STREAM, 0);
+    if (server_fd < 0) {
+        throw std::system_error(errno, std::generic_category(), "Failed to create server socket");
+    }
+
+    // Set socket options
+    int reuse = 1;
+    if (setsockopt(server_fd, SOL_SOCKET, SO_REUSEADDR, &reuse, sizeof(reuse)) < 0) {
```

Comment:
"  // // Since the tester restarts your program quite often, setting SO_REUSEADDR
  // // ensures that we don't run into 'Address already in use' errors"